pennylane/templates/layer.py

```diff
@@ -15,9 +15,27 @@
 Contains the ``layer`` template constructor.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import pennylane as qml
 from pennylane.templates.decorator import template as temp
 
-###################
+
+def _preprocess(args, depth):
+    """Validate and pre-process inputs."""
+
+    if not isinstance(depth, int):
+        raise ValueError("'depth' must be of type int, got {}".format(type(depth).__name__))
+
+    for arg in args:
+
+        if qml.tape_mode_active():
+            arg = qml.math.angle(arg)
+
```

Comment:
Else?

Reply:
Looking at the 'before', it doesn't seem to be needed 🤔 

@mariaschuld, can you remember why we need this in tape mode, but not in non-tape mode?